pytorch_lightning/utilities/fetching.py

```diff
@@ -236,19 +234,27 @@ def prefetching(self) -> None:
                 break
 
     def fetching_function(self) -> Tuple[Any, bool]:
+        assert self.dataloader_iter is not None
```

Comment:
Mind adding more comments on this method. This is quite core to the fetching and would benefit from line comments.